Jakarta - EE 10 - Support JAX-RS 3.1.0 in RESTEasy Reactive

JAX-RS 3.1.0 introduces some new methods and classes and we will need to implement them in RESTEasy Reactive.

How to reproduce:
- get the `jakarta-rewrite` branch from the upstream repository (this branch is already transformed to Jakarta) - it is rewritten daily (but these parts are not moving parts so it shouldn't be a problem)
- `mvn -Dquickly`
- `cd independent-projects/resteasy-reactive`
- change `jakarta.ws.rs-api.version` to `3.1.0` in `pom.xml`
- `mvn clean install`

First failure will be:

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project resteasy-reactive-common: Compilation failure
> [ERROR] /data/home/gsmet/git/quarkus-jakarta/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/jaxrs/RuntimeDelegateImpl.java:[30,8] org.jboss.resteasy.reactive.common.jaxrs.RuntimeDelegateImpl is not abstract and does not override abstract method createEntityPartBuilder(java.lang.String) in jakarta.ws.rs.ext.RuntimeDelegate
[ERROR]

Once this is done, I know there is another one related to the bootstrap and I think that's pretty much it - but I wasn't able to experiment fully.

Note: the tree won't function fully if you fix that because RESTEasy Classic is pointing to the EE 9 version for now but the idea is to have a compilable RESTEasy Reactive in `independent-projects` so that I can upgrade RESTEasy Classic and JAX-RS in the main bom _(I will handle that part)_.

I think the easiest way to work on that would be to create a branch in one's fork and push a commit on top of the `jakarta-rewrite` branch. 

Then I would get it and apply it on top of the transformation each night. And we would have an error if the patch doesn't apply cleanly but I don't think these areas change a lot. _(I will handle that part)_

These methods/classes have already been implemented in RESTEasy Classic 6.1.0.Final so that could be a source of inspiration.

Ideally, the `jakarta.ws.rs-api.version` upgrade will be done in a separate first commit as I think I will rely on OpenRewrite to change the version so that everything is centralized.

/cc @geoand

--- a/extensions/websockets/client/runtime/src/main/java/io/quarkus/websockets/BearerTokenClientEndpointConfigurator.java
+++ b/extensions/websockets/client/runtime/src/main/java/io/quarkus/websockets/BearerTokenClientEndpointConfigurator.java
@@ -4,6 +4,7 @@ import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
+import javax.net.ssl.SSLContext;
 import javax.websocket.ClientEndpointConfig;
 import javax.websocket.Decoder;
 import javax.websocket.Encoder;
@@ -52,5 +53,9 @@ public class BearerTokenClientEndpointConfigurator implements ClientEndpointConf
     @Override
     public Map<String, Object> getUserProperties() {
         return Collections.emptyMap();
+    }
+
+    public SSLContext getSSLContext() {
+        return null;
     }
 }

--- a/jakarta/rewrite.yml
+++ b/jakarta/rewrite.yml
@@ -134,7 +134,7 @@ recipeList:
   # Security - See below
   - org.openrewrite.maven.ChangePropertyValue:
       key: quarkus-security.version
-      newValue: 2.0.0.Alpha1
+      newValue: 2.0.0.Alpha2
   # Servlet
   - org.openrewrite.maven.ChangePropertyValue:
       key: jakarta.servlet-api.version

--- a/jakarta/rewrite.yml
+++ b/jakarta/rewrite.yml
@@ -345,15 +345,13 @@ recipeList:
       groupId: jakarta.authentication
       artifactId: jakarta.authentication-api
       version: "${jakarta.authentication-api}"
-  # TODO EE 10 2.1.0
   - org.openrewrite.maven.ChangePropertyValue:
       key: jakarta.authorization-api.version
-      newValue: 2.0.0
+      newValue: 2.1.0
       addIfMissing: true
-  # TODO EE 10 3.0.0
   - org.openrewrite.maven.ChangePropertyValue:
       key: jakarta.authentication-api
-      newValue: 2.0.0
+      newValue: 3.0.0
       addIfMissing: true
 ---
 type: specs.openrewrite.org/v1beta/recipe

--- a/jakarta/rewrite.yml
+++ b/jakarta/rewrite.yml
@@ -136,13 +136,12 @@ recipeList:
       key: quarkus-security.version
       newValue: 2.0.0.Alpha1
   # Servlet
-  # TODO EE 10 6.0.0
   - org.openrewrite.maven.ChangePropertyValue:
       key: jakarta.servlet-api.version
-      newValue: 5.0.0
+      newValue: 6.0.0
   - org.openrewrite.maven.ChangePropertyValue:
       key: quarkus-http.version
-      newValue: 5.0.0.Alpha1
+      newValue: 5.0.0.Alpha2
   # Transaction
   - org.openrewrite.maven.ChangePropertyValue:
       key: version.jta
@@ -171,10 +170,9 @@ recipeList:
       key: hibernate-validator.version
       newValue: 8.0.0.CR3
   # Websockets
-  # TODO EE 10 2.1.0
   - org.openrewrite.maven.ChangePropertyValue:
       key: jakarta.websocket-api.version
-      newValue: 2.0.0
+      newValue: 2.1.0
   # XML Bind
   - org.openrewrite.maven.ChangePropertyValue:
       key: jakarta.xml.bind-api.version

--- a/jakarta/rewrite.yml
+++ b/jakarta/rewrite.yml
@@ -72,7 +72,7 @@ recipeList:
   # JAX-RS - See below for the specs
   - org.openrewrite.maven.ChangePropertyValue:
       key: resteasy.version
-      newValue: 6.0.3.Final
+      newValue: 6.1.0.Final
   # JAXB - See below
   # JSON - See below
   # JSON Bind
@@ -205,7 +205,7 @@ recipeList:
       addIfMissing: true
   - org.openrewrite.maven.ChangePropertyValue:
       key: jakarta.ws.rs-api.version
-      newValue: 3.0.0
+      newValue: 3.1.0
       addIfMissing: true
 ---
 type: specs.openrewrite.org/v1beta/recipe

--- a/jakarta/transform.sh
+++ b/jakarta/transform.sh
@@ -351,6 +351,11 @@ git fetch origin jakarta-10-cdi
 JAKARTA_10_CDI_HASH=$(git rev-parse origin/jakarta-10-cdi)
 git cherry-pick -x ${JAKARTA_10_CDI_HASH}
 
+## JAX-RS/RESTEasy Reactive
+git fetch origin jakarta-10-jaxrs
+JAKARTA_10_JAXRS_HASH=$(git rev-parse origin/jakarta-10-jaxrs)
+git cherry-pick -x ${JAKARTA_10_JAXRS_HASH}
+
 # Build phase
 
 if [ "${REWRITE_TESTS_CONTAINERS-false}" == "true" ]; then

--- a/jakarta/transform.sh
+++ b/jakarta/transform.sh
@@ -292,6 +292,7 @@ sed -i 's@<excludedArtifact>jakarta.xml.bind:jakarta.xml.bind-api</excludedArtif
 sed -i 's@<excludedArtifact>jakarta.ws.rs:jakarta.ws.rs-api</excludedArtifact>@<excludedArtifact>org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_3.0_spec</excludedArtifact>\n                        <excludedArtifact>org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.1_spec</excludedArtifact>@g' extensions/resteasy-classic/resteasy-common/runtime/pom.xml
 rm extensions/resteasy-classic/resteasy-common/runtime/src/main/java/io/quarkus/resteasy/common/runtime/graal/{Target_org_jboss_resteasy_microprofile_config_FilterConfigSource.java,Target_org_jboss_resteasy_microprofile_config_ServletConfigSource.java,Target_org_jboss_resteasy_microprofile_config_ServletContextConfigSource.java,ServletMissing.java}
 sed -i 's@<quarkus.rest-client-reactive.scope>javax.enterprise.context.Dependent</quarkus.rest-client-reactive.scope>@<quarkus.rest-client-reactive.scope>jakarta.enterprise.context.Dependent</quarkus.rest-client-reactive.scope>@g' tcks/microprofile-rest-client-reactive/pom.xml
+sed -i '/public SSLContext getSSLContext() {/i \ \ \ \ @Override' extensions/websockets/client/runtime/src/main/java/io/quarkus/websockets/BearerTokenClientEndpointConfigurator.java
 
 sed -i 's@org.jboss.narayana.rts:narayana-lra@org.jboss.narayana.rts:narayana-lra-jakarta@g' extensions/narayana-lra/runtime/pom.xml
 sed -i 's@org.jboss.narayana.rts:lra-client@org.jboss.narayana.rts:lra-client-jakarta@g' extensions/narayana-lra/runtime/pom.xml

